rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -2223,6 +2224,58 @@ rocalRawCIFAR10Source(
     return output;
 }
 
+RocalTensor ROCAL_API_CALL
+rocalRawCIFAR10SourceSingleShard(
+    RocalContext p_context,
+    const char* source_path,
+    RocalImageColor rocal_color_format,
+    unsigned shard_id,
+    unsigned shard_count,
+    bool is_output,
+    bool shuffle,
+    bool loop,
+    unsigned out_width,
+    unsigned out_height,
+    const char* filename_prefix,
+    RocalShardingInfo rocal_sharding_info) {
+    Tensor* output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    try {
+        if (out_width == 0 || out_height == 0) {
```

Comment:
Without width and height can we not do this?

Reply:
Need width and height for creating tensor dims